config/src/keys.rs

```diff
@@ -88,7 +87,23 @@ pub struct NetworkKeyPairs {
     network_identity_public_key: X25519StaticPublicKey,
 }
 
-// required for serialization
+impl Clone for NetworkKeyPairs {
+    #[cfg(any(test, feature = "fuzzing"))]
+    fn clone(&self) -> Self {
+        Self {
+            network_signing_private_key: self.network_signing_private_key.clone(),
+            network_signing_public_key: self.network_signing_public_key.clone(),
+            network_identity_private_key: self.network_identity_private_key.clone(),
+            network_identity_public_key: self.network_identity_public_key.clone(),
+        }
+    }
+
+    #[cfg(not(any(test, feature = "fuzzing")))]
+    fn clone(&self) -> Self {
+        Self::default()
+    }
+}
+
```

Comment:
I think this is probably a not-so-good change. Anytime you do this sort of stuff in a Clone impl could lead to so very undesirable consequences. You might have the right keys and then you do a clone and now you don't have valid keys.

Reply:
Let me pop this commit out and see if anything bad happens. I think ideally we should be able to clone these configs and move toward using pin instead of doing this current PrivateKeyContainer stuff.